barretenberg/cpp/pil/avm/gadgets/range_check.pil

```diff
@@ -186,3 +187,30 @@ namespace range_check(256);
     #[LOOKUP_RNG_CHK_7]
     sel_rng_chk { u16_r7 } in main.sel_rng_16 { main.clk };
 
+    // ===== MEM TRACE RANGE CHECKS =====
+    pol commit mem_rng_chk;
+    // We range check 40 bits in the mem trace
+    mem_rng_chk * (rng_chk_bits - 40) = 0;
+
+    #[PERM_RNG_MEM]
+    mem_rng_chk {clk, value}
```

Comment:
I would rename clk with sthg like rng_idx because we use clk as the row index and/or the clock for the execution trace.